### ISSUE TITLE
Android build error using NativeScript Sidekick: app:mergeDebugResources

### ISSUE THREAD

**Issue by rchisholm** [NativeScript/NativeScript#6281]
### Did you verify this is a real problem by searching the [NativeScript Forum]

Yes

### Tell us about the problem

I'm building an app with NativeScript using TypeScript and Angular. This has happened twice while building the app.

In order to add resources (like images), I add them to my_app_folder\app\App_Resources\Android\src\main\res. I accidentally added a folder called "video" with a file called "sample_video.mp4" into the res folder. When I tried to run my app on my Android emulator using the cloud build, I received this error in my NativeScript Sidekick console:

 /mnt/storage/builds/_/b10d99b4991d936c4896cd72f0628901fa996d4c/4.2.4/myappname/platforms/android/app/src/main/res/video/sample_video.mp4: Error: The file name must end with .xml
I realized I had placed the file in the wrong place, deleted the video and the video folder, and tried to build the app again, but now I am consistently getting the above error, even though the offending file is long gone.

This has happened to me before (with a .png file with a capital letter in the name, which android does not like) and I would love to know why. Last time I had to delete my whole project, revert to my previous commit, and rebuild the app, which is frustrating.

My app ran perfectly fine before I added that file, so I know that the problem is not being caused by my code. I tried deleting the android platform and reinstalling it, to no avail.

Any help would be amazing as I am very confused as to why the file, which is no longer in my project, would continue to cause this problem.

### Which platform(s) does your issue occur on?
Android

### Please provide the following version numbers that your issue occurs with:
- CLI: 4.2.3

- Runtime(s):
    "tns-ios": {
      "version": "4.2.0"
    },
    "tns-android": {
      "version": "4.2.0"
    }
- Plugin(s):

  "dependencies": {
    "@angular/animations": "~6.1.0",
    "@angular/common": "~6.1.0",
    "@angular/compiler": "~6.1.0",
    "@angular/core": "~6.1.0",
    "@angular/forms": "~6.1.0",
    "@angular/http": "~6.1.0",
    "@angular/platform-browser": "~6.1.0",
    "@angular/platform-browser-dynamic": "~6.1.0",
    "@angular/router": "~6.1.0",
    "nativescript-angular": "~6.1.0",
    "nativescript-local-notifications": "^3.0.0",
    "nativescript-orientation": "2.2.1",
    "nativescript-theme-core": "~1.0.4",
    "nativescript-toast": "1.4.6",
    "nativescript-videoplayer": "4.2.0",
    "reflect-metadata": "~0.1.10",
    "rxjs": "~6.2.0",
    "tns-core-modules": "~4.2.0",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "codelyzer": "~4.3.0",
    "nativescript-dev-sass": "~1.6.0",
    "nativescript-dev-typescript": "~0.7.0",
    "nativescript-dev-webpack": "~0.15.0",
    "tslint": "~5.11.0",
    "typescript": "~2.7.2",
    "@angular/compiler-cli": "~6.1.0",
    "@ngtools/webpack": "~6.1.0"
  },

### Please tell us how to recreate the issue in as much detail as possible.
-create a new project in sidekick (any template will do)
-go into the folder "my_app_folder\app\App_Resources\Android\src\main\res" and create a new folder "video"
-place a mp4 file in that folder
-attempt a cloud build (debug) for android in sidekick, wait for it to fail
-go into the folder "my_app_folder\app\App_Resources\Android\src\main\res" and delete the folder "video"
-attempt a cloud build (debug) for android in sidekick

### Is there code involved? If so, please share the minimal amount of code needed to recreate the problem.
no

**Comment by rchisholm** [comment]
i should add, I also tried the option "Clean Cloud Workspace" in Sidekick but it did not solve the issue.

**Comment by rchisholm** [comment]
update: reverting to my previous commit does not solve the problem, so I don't know how to go forward from here. There must be some configuration stored in the cloud pertaining to my android build, but I don't know how to clean it out. In the past when I build Android apps, if I put an incorrect file in the res folder, I would get a similar error to this, but when I removed it, the error went away.

**Comment by rchisholm** [comment]
update: I was able to fix my build by deleting my app and cloning the repo into a different folder; I was then able to perform an Android cloud build.

 What's interesting is, after I cloned the repo into a new folder, I renamed the old one from "my_app_name_real" to "my_app_name_broken" and changed the newly cloned folder's name to "my_app_name" to "my_app_name_real" (the same name as the broken build) and then, when I tried to build it again, I got the same Android build error I've been having. So I had to change the name back to "my_app_name" and it built successfully. So the problematic cache (or whatever) is tied to my folder's name (and as a result, the name of the app in Sidekick).

Just when I think I'm getting the hang of this, some new hiccup comes up. It is a learning experience to be sure.
